lib/domain/usecases/atomic_swap_multi_buy.dart

```
@@ -0,0 +1,75 @@
+  final HttpConfig httpConfig;
+  final List<String> ids;
+  final String psbtHex;
+  final String buyerAddress;
+
+  const AtomicSwapMultiBuyParams({
+    required this.httpConfig,
+    required this.ids,
+    required this.psbtHex,
+    required this.buyerAddress,
+  });
+}
+
+class AtomicSwapMultiBuyUseCase
+    implements
+        UseCaseTE<List<AtomicSwapBuy>, AtomicSwapMultiBuyParams, String> {
+  final AtomicSwapRepository _atomicSwapRepository;
+  final ErrorService _errorService;
+  final AnalyticsService _analyticsService;
+
+  AtomicSwapMultiBuyUseCase({
+    AtomicSwapRepository? atomicSwapRepository,
+    ErrorService? errorService,
+    AnalyticsService? analyticsService,
+  })  : _atomicSwapRepository =
+            atomicSwapRepository ?? GetIt.I<AtomicSwapRepository>(),
+        _errorService = errorService ?? GetIt.I<ErrorService>(),
+        _analyticsService = analyticsService ?? GetIt.I<AnalyticsService>();
+
+  @override
+  TaskEither<String, List<AtomicSwapBuy>> call(
+      AtomicSwapMultiBuyParams params) {
+    final task = _atomicSwapRepository.atomicSwapMultiBuy(
+      httpConfig: params.httpConfig,
+      ids: params.ids,
+      psbtHex: params.psbtHex,
+      buyerAddress: params.buyerAddress,
+    );
+
+    return task.tap((success) {
+      _analyticsService.trackAnonymousEvent("atomic_swap_listing(s)_purchased");
+    }).tapError((error) {
+      _errorService.captureException(
+        error,
+        message: "Failed to execute atomic swap multi-buy",
+        context: error is NetworkError
+            ? {
+                "message": error.message,
+                "endpoint": error.endpoint,
+                "fullUrl": error.fullUrl,
+                "statusCode": error.statusCode,
+                "ids": params.ids,
+              }
+            : {
+                "message": error?.toString(),
+                "ids": params.ids,
+              },
+      );
+    }).mapLeft((error) => error.message);
```

is there a reason to mapLeft here? should we just return the error and let the caller do whatever?